### PR TITLE
Prove the universal property of Patch

### PR DESCRIPTION
**Pull request by ayberkt**
The proof is not complete yet but I'm opening up a draft PR for discussions.

### CODE REVIEW

**Comment by ayberkt** [comment]
Ready to review and merge now.

File: source/Locales/CompactRegular.lagda [hunk 1]
```diff
@@ -1098,6 +1191,99 @@ is-spectral-map F G (f , _) =
...
+
+\begin{code}
+
+compact-rel-syntax : (F : Frame 𝓤 𝓥 𝓦) → ⟨ F ⟩ → ⟨ F ⟩ → Ω (𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+compact-rel-syntax F U V =
+ Ɐ W ∶ ⟨ F ⟩ , is-compact-open F W ⇒ W ≤[ poset-of F ] U ⇒ W ≤[ poset-of F ] V
+
+syntax compact-rel-syntax F U V = U ≤ₖ[ F ] V
+
+spectral-yoneda : (F : Frame 𝓤 𝓥 𝓦) → is-spectral F holds → (U V : ⟨ F ⟩)
```
Comment (martinescardo) [inline source/Locales/CompactRegular.lagda:1253]: When a type is too long, it is better to put each arrow in a separate line, at the beginning.

File: source/Locales/PatchProperties.lagda [hunk 2]
```diff
@@ -642,6 +655,85 @@ module Complementation (X : Locale (𝓤 ⁺) 𝓤 𝓤) (σᴰ : spectralᴰ (
...
+I define below an alternative version of the module above due to a technical
+problem. Consider a spectral locale `A` with a proof `σ` of spectrality. When
+the module above is called with the data contained in `σ` (let's call this
+`σᴰ`), Agda does not recognise the fact that `σ = ∣ σᴰ ∣`. To circumvent this
+problem, I'm defining a version of this module that takes as argument the proof
+of spectrality instead of the structure contained within.
+
+\begin{code}
+
+module PatchComplementationAlternative (X : Locale (𝓤 ⁺) 𝓤 𝓤)
```
Comment (martinescardo) [inline source/Locales/PatchProperties.lagda:667]: This module already exists elsewhere, Ayberk tells me, and so should be deleted.

**Review by martinescardo — CHANGES_REQUESTED** [review]
Great! I've asked you to do a few changes right now, and then I will accept the pull request.

I've also asked you, verbally, to do other improvements, but this can be done more slowly, after the pull request is accepted.

File: source/Locales/CompactRegular.lagda (hunk 1 above)
Comment (ayberkt) [reply source/Locales/CompactRegular.lagda]: Done.

File: source/Locales/PatchProperties.lagda (hunk 2 above)
Comment (ayberkt) [reply source/Locales/PatchProperties.lagda:667]: I'll do this later.

**Review by martinescardo — APPROVED** [review]

**Pull request merged**